File: Api/Employees/Models/Employee.cs
using Api.ApprovalRequests.Models;
using Api.Employees.Enums;
using Api.LeaveRequests.Models;
using Api.Projects.Models;

namespace Api.Employees.Models;

public record Employee
{
    public int Id { get; init; }

    public required string FullName { get; init; }

    public required string Subdivision { get; init; }

    public EmployeePosition Position { get; init; }

    public EmployeeStatus Status { get; init; }

    public int OutOfOfficeBalance { get; init; }

    public required EmployeeBrief PeoplePartner { get; init; }
}

File: client/src/employees/models/Employee.ts
import LeaveRequest from "src/leaveRequests/models/LeaveRequest";
import Project from "src/projects/models/Project";

type Employee = {
  id: number;
  fullName: string;
  subdivision: string;
  position: string;
  status: string;
  outOfOfficeBalance: number;
  peoplePartner?: Employee;
  projects: Project[];
  managedProjects: Project[];
  employees: Employee[];
  leaveRequests: LeaveRequest[];
}

export default Employee;